src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -56,12 +69,65 @@ private class ConfigurationCacheItem
 
         public Task<bool> IsEnabledAsync(string feature)
         {
-            return IsEnabledAsync<object>(feature, null, false);
+            return IsEnabledWithVariantsAsync<object>(feature, appContext: null, useAppContext: false, CancellationToken.None);
         }
 
         public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext)
         {
-            return IsEnabledAsync(feature, appContext, true);
+            return IsEnabledWithVariantsAsync(feature, appContext, useAppContext: true, CancellationToken.None);
+        }
+
+        public Task<bool> IsEnabledAsync(string feature, CancellationToken cancellationToken)
+        {
+            return IsEnabledWithVariantsAsync<object>(feature, appContext: null, useAppContext: false, cancellationToken);
+        }
+
+        public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext, CancellationToken cancellationToken)
+        {
+            return IsEnabledWithVariantsAsync(feature, appContext, useAppContext: true, cancellationToken);
+        }
+
+        private async Task<bool> IsEnabledWithVariantsAsync<TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        {
+            bool isFeatureEnabled = await IsEnabledPrivateAsync(feature, appContext, useAppContext, cancellationToken).ConfigureAwait(false);
+
+            VariantDefinition variantDefinition;
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider.GetFeatureDefinitionAsync(feature).ConfigureAwait(false);
+            TargetingContext targetingContext;
+
+            if (featureDefinition == null || !(featureDefinition.Variants?.Any() ?? false) || featureDefinition.Status == FeatureStatus.Disabled)
```

Comment:
`featureDefinition.Status == FeatureStatus.Disabled` If that's the case we should be returning false.

Reply:
Same with non-existent feature definition.

Reply:
I see, I was assuming that if those were true then `IsEnabled` would return false anyway. I can return false explicitly